src/wallet/rpc/backup.cpp

```diff
@@ -406,842 +122,6 @@ RPCHelpMan removeprunedfunds()
-            if (import_data.witnessscript) warnings.push_back("Ignoring witnessscript as this is not a (P2SH-)P2WSH script.");
-            for (auto it = privkey_map.begin(); it != privkey_map.end(); ) {
-                auto oldit = it++;
-                if (import_data.used_keys.count(oldit->first) == 0) {
-                    warnings.push_back("Ignoring irrelevant private key.");
-                    privkey_map.erase(oldit);
-                }
-            }
-            for (auto it = pubkey_map.begin(); it != pubkey_map.end(); ) {
-                auto oldit = it++;
-                auto key_data_it = import_data.used_keys.find(oldit->first);
-                if (key_data_it == import_data.used_keys.end() || !key_data_it->second) {
-                    warnings.push_back("Ignoring public key \"" + HexStr(oldit->first) + "\" as it doesn't appear inside P2PKH or P2WPKH.");
-                    pubkey_map.erase(oldit);
-                }
-            }
-        }
-    }
-
-    return warnings;
-}
-
-static UniValue ProcessImportDescriptor(ImportData& import_data, std::map<CKeyID, CPubKey>& pubkey_map, std::map<CKeyID, CKey>& privkey_map, std::set<CScript>& script_pub_keys, bool& have_solving_data, const UniValue& data, std::vector<std::pair<CKeyID, bool>>& ordered_pubkeys)
-{
-    UniValue warnings(UniValue::VARR);
-
-    const std::string& descriptor = data["desc"].get_str();
-    FlatSigningProvider keys;
-    std::string error;
-    auto parsed_descs = Parse(descriptor, keys, error, /* require_checksum = */ true);
-    if (parsed_descs.empty()) {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, error);
-    }
-    if (parsed_descs.at(0)->GetOutputType() == OutputType::BECH32M) {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Bech32m descriptors cannot be imported into legacy wallets");
-    }
-
-    std::optional<bool> internal;
-    if (data.exists("internal")) {
-        if (parsed_descs.size() > 1) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Cannot have multipath descriptor while also specifying \'internal\'");
-        }
-        internal = data["internal"].get_bool();
-    }
-
-    have_solving_data = parsed_descs.at(0)->IsSolvable();
-    const bool watch_only = data.exists("watchonly") ? data["watchonly"].get_bool() : false;
-
-    int64_t range_start = 0, range_end = 0;
-    if (!parsed_descs.at(0)->IsRange() && data.exists("range")) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Range should not be specified for an un-ranged descriptor");
-    } else if (parsed_descs.at(0)->IsRange()) {
-        if (!data.exists("range")) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Descriptor is ranged, please specify the range");
-        }
-        std::tie(range_start, range_end) = ParseDescriptorRange(data["range"]);
-    }
-
-    // Only single key descriptors are allowed to be imported to a legacy wallet's keypool
-    bool can_keypool = parsed_descs.at(0)->IsSingleKey();
```

Comment:
In #31243, `IsSingleKey` was added with a TODO to remove it after removing legacy wallets.
I checked that no other usage of `IsSingleKey` is left, it can be removed. 

https://github.com/bitcoin/bitcoin/blob/6d5edfcc585bab3374ae14aa7918b3e178e016aa/src/script/descriptor.h#L114-L117